specs/eip4844/p2p-interface.md

```diff
@@ -129,18 +130,45 @@ Per `context = compute_fork_digest(fork_version, genesis_validators_root)`:
 
 **Protocol ID:** `/eth2/beacon_chain/req/beacon_blocks_by_root/2/`
 
-The EIP-4844 fork-digest is introduced to the `context` enum to specify EIP-4844 beacon block type.
+`BeaconBlocksByRoot v2` will be disabled in favor of `BeaconBlockAndBlobsSidecarByRoot v1` for tighter coupling of blocks and sidecars.
 
-Per `context = compute_fork_digest(fork_version, genesis_validators_root)`:
+Clients MUST disable `BeaconBlocksByRoot v2` after `EIP4844_FORK_EPOCH`.
 
-[1]: # (eth2spec: skip)
+#### BeaconBlockAndBlobsSidecarByRoot v1
 
-| `fork_version`           | Chunk SSZ type             |
-| ------------------------ | -------------------------- |
-| `GENESIS_FORK_VERSION`   | `phase0.SignedBeaconBlock` |
-| `ALTAIR_FORK_VERSION`    | `altair.SignedBeaconBlock` |
-| `BELLATRIX_FORK_VERSION` | `bellatrix.SignedBeaconBlock` |
-| `EIP4844_FORK_VERSION`   | `eip4844.SignedBeaconBlock`   |
+**Protocol ID:** `/eth2/beacon_chain/req/beacon_block_and_blobs_sidecar_by_root/1/`
+
+Request Content:
+
+```
+(
+  List[Root, MAX_REQUEST_BLOCKS]
+)
+```
+
+Response Content:
+
+```
+(
+  List[SignedBeaconBlockAndBlobsSidecar, MAX_REQUEST_BLOCKS]
```

Comment:
Should I rename `MAX_REQUEST_BLOCKS` to include sidecars? I didn't feel strongly about this

Reply:
Fine by me. Though another thing is do we wanna tweak the current `MAX_REQUEST_BLOCKS` value? If a client requests the max and they all contain sidecars, that's a huge amount of data transmitted.

Reply:
Unless this is also constrained by the `MAX_CHUNK_SIZE`.

Reply:
I think it is clear as it is

Reply:
I do think that we should at least consider `MAX_REQUEST_BLOCKS` for this and range requests given the size changes. That said, you only *MUST* respond with 1 in a given request/range so even if the max stays high, clients can down regulate responses and stay to spec